src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -4254,5 +4255,24 @@ GUI_TEST_CLASS_DEFINITION(test_7806) {
     CHECK_SET_ERR(size == 4, "chrM.fa in SAM dir is changed, size: " + QString::number(size));
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7830) {
+    // 1. Open COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW/", "COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // 2. Right button-> Statistics-> Generate distance matrix
+    // 3. Click "Generate"
+    GTUtilsDialog::add(os, new PopupChooser(os, { MSAE_MENU_STATISTICS, "Generate distance matrix" }, GTGlobals::UseMouse));
+    GTUtilsDialog::add(os, new DistanceMatrixDialogFiller(os, true, true, true));
+    GTMenu::showContextMenu(os, GTUtilsMdi::activeWindow(os));
+
+    // Expected: legend exists
+    QTextBrowser* v = GTUtilsMdi::activeWindow(os)->findChild<QTextBrowser*>();
+    QString text = v->toHtml();
+    CHECK_SET_ERR(text.contains("Legend"), text);
```

Comment:
Add more checks for the values & colors in the matrix, so the test will check the actual results.

Reply:
I rewrote this test a bit. Now distance matrix is saved to sandbox and compares with the example, which is in the test repo.